lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2424,26 +2424,31 @@ CallInst *SPIRVToLLVM::transFixedPointInst(SPIRVInstruction *BI,
   // Literal S Literal I Literal rI Literal Q Literal O
 
   Type *RetTy = transType(BI->getType());
+
   auto Inst = static_cast<SPIRVFixedPointIntelInst *>(BI);
+  Type *InTy = transType(Inst->getOperand(1)->getType());
 
   IntegerType *Int32Ty = IntegerType::get(*Context, 32);
   IntegerType *Int1Ty = IntegerType::get(*Context, 1);
 
-  SmallVector<Type *, 7> ArgTys = {transType(Inst->getOperand(1)->getType()),
-                                   Int1Ty,
-                                   Int32Ty,
-                                   Int32Ty,
-                                   Int32Ty,
-                                   Int32Ty};
-
+  SmallVector<Type *, 7> ArgTys = {InTy,    Int1Ty,  Int32Ty,
+                                   Int32Ty, Int32Ty, Int32Ty};
   FunctionType *FT = FunctionType::get(RetTy, ArgTys, false);
 
+  std::stringstream Suffix;
+  Suffix << ".i" << RetTy->getIntegerBitWidth() << ".i"
+         << InTy->getIntegerBitWidth();
+
   Op OpCode = Inst->getOpCode();
-  Function *Func = Function::Create(FT, GlobalValue::ExternalLinkage,
-                                    SPIRVFixedPointIntelMap::rmap(OpCode), M);
-  Func->setCallingConv(CallingConv::SPIR_FUNC);
-  if (isFuncNoUnwind())
-    Func->addFnAttr(Attribute::NoUnwind);
+  std::string FuncName = SPIRVFixedPointIntelMap::rmap(OpCode) + Suffix.str();
+
+  FunctionCallee FCallee = M->getOrInsertFunction(FuncName, FT);
+
+  if (Function *Fn = dyn_cast<Function>(FCallee.getCallee())) {
```

Comment:
Should we check if this is Function*? What if this is not so?

Reply:
We check that it is a `Function*` with `dyn_cast<Function>`. Probably we might want to check that `FCallee.getCallee()` doesn't return `nullptr`. We should use `dyn_cast_or_null` in this case.

Reply:
Well, actually we expect that it will be always Function* because we insert function call on the line above.
I would rather use just `cast<Function>` call without an if  check.